src/main/java/com/cloudbees/jenkins/support/impl/SlaveCommandStatistics.java

```diff
@@ -56,7 +65,14 @@
 @Extension
 public final class SlaveCommandStatistics extends Component {
 
+    /*protected*/ static @Nonnegative int MAX_STATS_SIZE = 1000;
+
+    private final Object statLock = new Object();
+    @GuardedBy("statLock")
     private final Map<String, Statistics> statistics = Collections.synchronizedSortedMap(new TreeMap<>());
```

Comment:
Presumably do not need the synchronization wrapper if you are explicitly synchronizing anyway?

Reply:
Yeah, I guess we can get rid of that now.